src/main/java/com/archisacademy/jobportal/services/impl/ProjectServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.archisacademy.jobportal.services.impl;
+
+import com.archisacademy.jobportal.dto.ProjectDto;
+import com.archisacademy.jobportal.loggers.MainLogger;
+import com.archisacademy.jobportal.loggers.messages.ProjectMessage;
+import com.archisacademy.jobportal.mapper.ProjectMapper;
+import com.archisacademy.jobportal.model.Project;
+import com.archisacademy.jobportal.repositories.ProfileRepository;
+import com.archisacademy.jobportal.repositories.ProjectRepository;
+import com.archisacademy.jobportal.services.ProjectService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProjectServiceImpl implements ProjectService {
+    private final ProjectRepository projectRepository;
+    private final ProfileRepository profileRepository;
+    private final ProjectMapper projectMapper;
+    private final static MainLogger LOGGER = new MainLogger(ProjectServiceImpl.class);
+
+    public ProjectServiceImpl(ProjectRepository projectRepository, ProfileRepository profileRepository, ProjectMapper projectMapper) {
+        this.projectRepository = projectRepository;
+        this.profileRepository = profileRepository;
+        this.projectMapper = projectMapper;
+    }
+
+    @Override
+    @Transactional
+    public String createProject(ProjectDto projectDto) {
+        Project project = projectMapper.toEntity(projectDto);
+        project.setProfile(profileRepository.findById(projectDto.getProfileId())
+                .orElseThrow(() -> {
+                    LOGGER.log(ProjectMessage.PROFILE_NOT_FOUND + projectDto.getProfileId(), HttpStatus.NOT_FOUND);
+                    return null;
+                }));
+        projectRepository.save(project);
+        return ProjectMessage.PROJECT_CREATED_SUCCESS;
+    }
+
+    @Override
+    @Transactional
+    public String deleteProject(Long id) {
+        Project project = projectRepository.findById(id)
+                .orElseThrow(() -> {
+                    LOGGER.log(ProjectMessage.PROJECT_NOT_FOUND + id, HttpStatus.NOT_FOUND);
+                    return null;
+                });
+
+        projectRepository.deleteById(id);
+
+        LOGGER.log(String.format(ProjectMessage.PROJECT_DELETED_LOG, id), HttpStatus.OK);
+        return ProjectMessage.PROJECT_DELETED_SUCCESS;
+    }
+
+    @Override
+    @Transactional
+    public String updateProject(Long projectId, ProjectDto projectDto) {
+        Project existingProject = projectRepository.findById(projectId)
```

Comment:
again same thing like the Furkan's PR please do create a method to fetch project by its id and use it in all the methods you want.